### PR TITLE
merge Feature/finalgame into develop

### PR DESCRIPTION
**Pull request by yaakoubichaima**
..

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,64 +1,61 @@
...
-        this.setState({ playersSubmitted: true });
-        event.preventDefault();
-      }
+  constructor(props) {
+    super(props);
+    this.state = {
+      player1: '',
+      player2: '',
+      playersSubmitted: false,
+      score: [0, 0]
```
Comment (moezF) [inline src/App.js:12]: score here is never used, remove it from state

File: src/Board.js [hunk 2]
```diff
@@ -3,19 +3,14 @@ import React from 'react'
...
-         squares: Array(9).fill(null),
-         xIsNext: true,
-      };
-     } */
   renderSquare(i, isValid) {
     return (
       <div>
         <div>
-          {((this.props.validSquares).indexOf(i) !== -1)
+          { (this.props.validSquares)
```
Comment (moezF) [inline src/Board.js:10]: never use composed ternary operator use if else instead

File: src/Game.js [hunk 3]
```diff
@@ -54,17 +57,24 @@ export class Game extends React.Component {
...
-      {
-         //this.setState({validSqaures: lines[i]}) 
-          return squares[a];
+      if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
+        if (squares[a] == 'X') {
+          this.setState({ winner: 'X' })
+          this.setState({ scoreX: this.state.scoreX + 1 })
+        }
+        else {
+          if (squares[a] == 'O') {
```
Comment (moezF) [inline src/Game.js:66]: no need for the second if,
if there is a winner and it's not X it's obviously O

File: src/Game.js [hunk 4]
```diff
@@ -54,17 +57,24 @@ export class Game extends React.Component {
...
+            this.setState({ winner: 'O' })
+            this.setState({ scoreO: this.state.scoreO + 1 })
+          }
+        }
+        return;
       }
     }
-    return null;
+    //return null;
```
Comment (moezF) [inline src/Game.js:74]: remove comment

File: src/Game.js [hunk 5]
```diff
@@ -54,17 +57,24 @@ export class Game extends React.Component {
...
   }
 
-  getValidline(squares)
-  {
+  getValidline(squares) {
```
Comment (moezF) [inline src/Game.js:77]: rename to getValidLine

**Review by moezF — CHANGES_REQUESTED** [review]

File: src/Board.js [hunk 6]
```diff
@@ -3,21 +3,24 @@ import React from 'react'
...
-      this.state = {
-         squares: Array(9).fill(null),
-         xIsNext: true,
-      };
-     } */
+  checkValid(i){
+          let isValid
+          if(this.props.validSquares)
+              if((this.props.validSquares).indexOf(i) !== -1)
+              {isValid = true}
```
Comment (moezF) [inline src/Board.js:10]: missing indentation

File: src/Game.js [hunk 7]
```diff
@@ -16,31 +16,33 @@ export class Game extends React.Component {
       stepNumber: 0,
       xIsNext: true,
       restart: false,
+      winner: null,
+      scoreX: 0,
+      scoreO: 0
     }
-    //let l = Array(9).fill(null)
-    this.restart = this.restart.bind(this)
+    //this.restart = this.restart.bind(this)
```
Comment (moezF) [inline src/Game.js:23]: remove comment

**Review by moezF — CHANGES_REQUESTED** [review]

**Comment by moezF** [comment]
@yaakoubichaima  use meaningful commit messages

File: src/Game.js (hunk 7 above)
Comment (yaakoubichaima) [reply src/Game.js:23]: I left these comments just to remember to ask you about '.bind(this)' later

**Comment by yaakoubichaima** [comment]
> @yaakoubichaima use meaningful commit messages (no need to change the message, it's complicated to change it)

Yeah sure, It was a mistake and I didn't know how to change it.

File: src/Game.js (hunk 7 above)
Comment (moezF) [reply src/Game.js]: about that read this article: https://www.freecodecamp.org/news/the-complete-guide-to-this-in-javascript/

**Review by moezF — APPROVED** [review]

**Pull request merged**